src/devices/emax.c

```diff
@@ -0,0 +1,233 @@
+
+            decoder_output_data(decoder, data);
+            return 1;
+        }
+        else {  // if Rain/Wind sensor
+        
+            int temp_raw      = ((b[4] & 0x0f) << 8) | (b[5]); // weird format
+            float temp_f      = (temp_raw - 900) * 0.1f;
+            int humidity      = b[6];
+            int wind_raw      = ((b[7] - 1) << 8 ) | (b[8] -1);   // all default is 01 so need to remove 1 from all bytes.
+            float speed_kmh     = wind_raw * 0.2f;        
+            int direction_deg = (((b[9] & 0x0f) - 1) << 8) | (b[10] - 1);
+            int rain_raw      = ((b[11] - 1) << 8 ) | (b[12] -1);
+            float rain_mm     = rain_raw * 0.2f;            
+            int uv_index      = (b[13] & 0x0f) - 1;
+            int lux_multi     = ((b[14] & 0x80) >> 7);
+            int light_lux     = ((b[14] & 0x7f) - 1) << 8 | (b[15] - 1);
+            if (lux_multi == 1) {
+                light_lux = light_lux * 10;
+            }
+            
+            /* clang-format off */
+            data_t *data = data_make(
+                    "model",            "",                 DATA_STRING, "EMAX Temp Hum Rain Wind UV Lux Sensor",
+                    "id",               "",                 DATA_FORMAT, "%03x", DATA_INT,    id,
+                    "channel",          "Channel",          DATA_INT,    channel,
+                    "battery_ok",       "Battery_OK",       DATA_INT,    !battery_low,
+                    "temperature_F",    "Temperature_F",    DATA_FORMAT, "%.1f", DATA_DOUBLE, temp_f,
+                    "humidity",         "Humidity",         DATA_FORMAT, "%u",   DATA_INT,    humidity,
+                    "wind_avg_km_h",    "Wind avg speed",   DATA_FORMAT, "%.1f km/h",  DATA_DOUBLE, speed_kmh,
+                    "wind_dir_deg",     "Wind Direction",   DATA_INT,    direction_deg,
+                    "rain_mm",          "Total rainfall",   DATA_FORMAT, "%.1f mm",  DATA_DOUBLE, rain_mm,
+                    "uv",               "UV Index",         DATA_FORMAT, "%u",       DATA_INT,    uv_index,
+	                "light_lux",        "Lux",              DATA_FORMAT, "%u",       DATA_INT,    light_lux,
+                    "pairing",          "Pairing?",         DATA_COND,   pairing,    DATA_INT,    !!pairing,
+                    "mic",              "Integrity",        DATA_STRING, "CHECKSUM",
+                    NULL);
+            /* clang-format on */
+
+            decoder_output_data(decoder, data);
+            return 1;
+                        
+        }
+    }
+    return ret;
+}
+
+static char *output_fields[] = {
+        "model",
+        "id",
+        "channel",
+        "battery_ok",
+        "temperature_F",
+        "humidity",
+        "wind_avg_km_h",
+        "rain_mm",
+        "wind_dir_deg",
+        "uv",
+        "light_lux",
+        "pairing",
+        "mic",
+        NULL,
+};
+
+r_device emax = {
+        .name        = "EMAX Weather sensor",
```

Comment:
This should be "EMAX EM3390W6 / Altronics X7064 weather sensor",
(people might otherwise wonder "where did the Altronics go?)

Reply:
Hi,  thanks for your support.

Agree, the reason why I had left the older decoder file.
I found also several references for the same Weather Station. 
For sure for Altronics : x7064 = simple Temp/Hum sensor  and x7063 = 5-in-1 Rain/Wind/.... sensor : 
https://www.altronics.com.au/p/x7063-touchscreen-wifi-wireless-home-weather-station/

For Emax , EM3390W6 is the brand reference into the Optex 990040 user guide in certification section.
I found also for Emax, W6, WEC-W6, 3390TX (W6) manuals.
Infactory version is also not clear, the Weather Station is called FWS-1200, but the reference is NX6396-944.

So could be "Emax W6 and rebrand Altronics x7063/4, Optex SM-040, Infactory FWS-1200, Newentor Q9 Weather sensor"

Reply:
Yes, that's a very good text to inform other people.